src/test/scala/com/bwsw/cloudstack/entities/common/JsonMapperTestSuite.scala

```diff
@@ -0,0 +1,91 @@
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package com.bwsw.cloudstack.entities.common
+
+import com.fasterxml.jackson.databind.JsonMappingException
+import org.scalatest.{FlatSpec, PrivateMethodTester}
+
+class JsonMapperTestSuite extends FlatSpec with PrivateMethodTester {
+  val firstValue = "1"
+  val secondValue = "2"
+
+  "deserialize" should "deserialize json string to simple case class if setIgnoreUnknownProperties flag is true" +
+    "and Json string has equals fields with the class" in {
+    val testJson = "{\"first\":\"" + s"$firstValue" + "\",\"second\":\"" + s"$secondValue" + "\"}"
+    val expectedEntity = Test.TestEntity(firstValue, secondValue)
+    val jsonMapper = new JsonMapper(true)
+    assert(jsonMapper.deserialize[Test.TestEntity](testJson) == expectedEntity)
+  }
+
+  "deserialize" should "deserialize json string to simple case class if setIgnoreUnknownProperties flag is true" +
+    "and Json string has non equals fields with the class" in {
+    val testJson = "{\"first\":\"" + s"$firstValue" + "\",\"second\":\"" + s"$secondValue" + "\",\"third\":\"3\"}"
+    val expectedEntity = Test.TestEntity(firstValue, secondValue)
+    val jsonMapper = new JsonMapper(true)
+
+    assert(jsonMapper.deserialize[Test.TestEntity](testJson) == expectedEntity)
+  }
+
+  "deserialize" should "deserialize json string to simple case class if setIgnoreUnknownProperties flag is false" +
+    "and Json string has equals fields with the class" in {
+    val testJson = "{\"first\":\"" + s"$firstValue" + "\",\"second\":\"" + s"$secondValue" + "\"}"
+    val expectedEntity = Test.TestEntity(firstValue, secondValue)
+    val jsonMapper = new JsonMapper(false)
+
+    assert(jsonMapper.deserialize[Test.TestEntity](testJson) == expectedEntity)
+  }
+
+  "deserialize" should "deserialize json string to parametrized case class if setIgnoreUnknownProperties flag is false" +
+    "and Json string has equals fields with the class" in {
+    val testJson = "{\"first\":\"" + s"$firstValue" + "\",\"second\":\"" + s"$secondValue" + "\"}"
+    val expectedEntity = Test.ParametrizedTestEntity[String](firstValue, secondValue)
+    val jsonMapper = new JsonMapper(false)
+
+    assert(jsonMapper.deserialize[Test.ParametrizedTestEntity[String]](testJson) == expectedEntity)
+  }
+
+  "deserialize" should "throw JsonMappingException if setIgnoreUnknownProperties flag is false" +
+    "and Json string has non equals fields with the target entity" in {
+    val testJson = "{\"first\":\"" + s"$firstValue" + "\",\"second\":\"" + s"$secondValue" + "\",\"third\":\"3\"}"
+    val expectedEntity = Test.TestEntity(firstValue, secondValue)
+    val jsonMapper = new JsonMapper(false)
+
+    assertThrows[JsonMappingException](jsonMapper.deserialize[Test.TestEntity](testJson))
+  }
+
+  "serialize" should "serialize entity to Json string" in {
```

Comment:
"serialize" should "convert an entity to a json string"

Reply:
ok